src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/ShiftInstruction.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Linq.Expressions.Interpreter
+{
+    internal abstract class ShiftInstruction : Instruction
+    {
+        public sealed override int ConsumedStack => 2;
+
+        public sealed override int ProducedStack => 1;
+
+        protected abstract object Shift(object value, object shift);
+
+        public sealed override int Run(InterpretedFrame frame)
+        {
+            object shift = frame.Pop();
+            object value = frame.Pop();
+            frame.Push(value == null | shift == null ? null : Shift(value, shift));
+            return +1;
```

Comment:
@bartdesmet @VSadov @hughbe what do you think of this change (which could be extended to other operator instructions)? At the time I committed it I was thinking of it as a straight-forward refactoring of commonality, but it does introduce another virtual call, so it's not without cost, and I'm not torn on it.